src/replication/DealReplicationWorker.ts

```diff
@@ -595,4 +595,27 @@ export default class DealReplicationWorker extends BaseService {
       }
     );
   }
+
+  private async currentBlockHeight() : Promise<number> {
+    try {
+      return await this.lotusBlockHeight()
+    } catch (error) {
+      this.logger.warn(`get lotus block height failed`, error);
+    }
+    return HeightFromCurrentTime();
+  }
+
+  private async lotusBlockHeight (): Promise<number> {
+    const statusCommand = `${this.lotusCMD} status`
```

Comment:
Can we use API calls for this? We are trying to steer away from any direct cli calls.

Reply:
edited PR